File: app/src/main/java/com/borisruzanov/russianwives/mvp/model/interactor/ChatMessageInteractor.java
package com.borisruzanov.russianwives.mvp.model.interactor;

import android.net.Uri;

import com.borisruzanov.russianwives.mvp.model.repository.FirebaseRepository;
import com.borisruzanov.russianwives.utils.UpdateCallback;

public class ChatMessageInteractor {

    private FirebaseRepository repository;

    public ChatMessageInteractor(FirebaseRepository repository) {
        this.repository = repository;
    }

    public void initChat(String friendUid){
        repository.initChat(friendUid);
    }

    public void sendMessage(String friendUid, String message, UpdateCallback callback){
        repository.sendMessage(friendUid, message, callback);
    }

    public void sendImage(String friendUid, Uri imageUri, UpdateCallback callback){
        repository.sendImage(friendUid, imageUri, callback);
    }

}

File: app/src/main/java/com/borisruzanov/russianwives/mvp/model/interactor/chatmessage/ChatMessageInteractor.kt
package com.borisruzanov.russianwives.mvp.model.interactor.chatmessage

import android.net.Uri

import com.borisruzanov.russianwives.mvp.model.repository.chatmessage.ChatMessageRepository
import com.borisruzanov.russianwives.utils.UpdateCallback
import javax.inject.Inject

class ChatMessageInteractor @Inject constructor(private val repository: ChatMessageRepository) {

    fun initChat(friendUid: String) {
        repository.initChat(friendUid)
    }

    fun sendMessage(friendUid: String, message: String, callback: UpdateCallback) {
        repository.sendMessage(friendUid, message, callback)
    }

    fun sendImage(friendUid: String, imageUri: Uri, callback: UpdateCallback) {
        repository.sendImage(friendUid, imageUri, callback)
    }

    fun removeOldMessage(chatUser: String?, id: String, message: String) {
        repository.removeOldMessage(chatUser,id,message)
    }

}
